tests/std/tests/P0088R3_variant_msvc/test.cpp

```diff
@@ -741,6 +741,87 @@ namespace msvc {
         }
     } // namespace gh2770
 
+    namespace gh4901 {
+#if _HAS_CXX20
+#define CONSTEXPR20 constexpr
+#else // ^^^ _HAS_CXX20 / !_HAS_CXX20 vvv
+#define CONSTEXPR20 inline
+#endif // ^^^ !_HAS_CXX20  ^^^
+        struct X {
+            CONSTEXPR20 ~X() {}
+        };
+
+        struct Y {
+            X _;
+        };
+
+        struct ZA {
+            std::variant<Y, int> z;
+        };
+
+        struct ZB {
+            std::variant<int, Y> z;
+        };
+
+#if _HAS_CXX20
+        static_assert(ZA{0}.z.index() == 1);
+        static_assert(ZA{Y{}}.z.index() == 0);
+        static_assert(ZB{0}.z.index() == 0);
+        static_assert(ZB{Y{}}.z.index() == 1);
+#endif // _HAS_CXX20
+
+        static_assert(std::is_nothrow_destructible_v<X>);
+        static_assert(std::is_nothrow_destructible_v<Y>);
+        static_assert(std::is_nothrow_destructible_v<std::variant<Y, int>>);
+        static_assert(std::is_nothrow_destructible_v<std::variant<int, Y>>);
+        static_assert(std::is_nothrow_destructible_v<ZA>);
+        static_assert(std::is_nothrow_destructible_v<ZB>);
+
+        // Verify that variant::~variant is always noexcept, per N4988 [res.on.exception.handling]/3.

```

Comment:
"always" implies a universal quantification which this test doesn't actually verify, and probably can't.
```suggestion
        // Verify that variant::~variant is noexcept even when an alternative has a potentially-throwing destructor,
        // per N4988 [res.on.exception.handling]/3.
```